admin-ui/src/components/Footer/Footer.js

```diff
@@ -1,15 +1,27 @@
 import React, { Component } from 'react'
 import { connect } from 'react-redux'
 
+
 class Footer extends Component {
-  render () {
+  constructor(props) {
+    super(props)
+    this.state = {
+    }
+  }
+
+  render() {
     return (
       <footer className='app-footer'>
         <span>
           <a href='https://github.com/SignalK/signalk-server-node/'>
             Signal K Server
-          </a>
+          </a>      
         </span>
+        {(this.props.loginStatus.status === 'notLoggedIn' || this.props.loginStatus.status === 'loggedIn' ) && (
```

Comment:
Without the test The page failed to load with an undefined error for
{this.props.serverSpecification.server.version}

Im not sure why this is the case, but as soon as I put the test in place it seemed to force the props to be 'defined'.

So perhaps my error is elsewhere?

Reply:
When the ui first loads and renders there is no server version yet loaded. You should add a test here so that if there is no this.props.serverSpecification you render an empty string or just -.

Reply:
My test for props being undefined:
{this.props.serverSpecification === undefined && (
          <span>
            &nbsp; version {this.props.serverSpecification.server.version} 
          </span>
        )}
works ok, but if I try to test for !==undefined it fails to find the props children and I get this console error:
{this.props.serverSpecification !== undefined && (
          <span>
            &nbsp; version {this.props.serverSpecification.server.version} 
          </span>
        )}

Uncaught TypeError: Cannot read property 'version' of undefined

Reply:
should be `typeof this.props.serverPecification !== undefined`

Reply:
Add quotes around undefined in Scott’s code